ui_curses.c

```diff
@@ -1794,40 +1794,38 @@ static void clear_error(void)
 
 /* screen updates }}} */
 
-static void spawn_status_program(void)
+static void spawn_status_program_inner(const char *status_text, struct track_info *ti)
 {
-	enum player_status status;
-	const char *stream_title = NULL;
-	char *argv[32];
-	int i;
-
 	if (status_display_program == NULL || status_display_program[0] == 0)
 		return;
 
-	status = player_info.status;
-	if (status == PLAYER_STATUS_PLAYING && player_info.ti && is_http_url(player_info.ti->filename))
-		stream_title = get_stream_title();
+	char *argv[32];
+	int i = 0;
 
-	i = 0;
 	argv[i++] = xstrdup(status_display_program);
 
 	argv[i++] = xstrdup("status");
-	argv[i++] = xstrdup(player_status_names[status]);
-	if (player_info.ti) {
+	argv[i++] = xstrdup(status_text);
+
+	if (ti) {
```

Comment:
please extract this to a seperate function to keep the line count per function resonable.